chia/util/config.py

```diff
@@ -325,3 +327,18 @@ def load_defaults_for_missing_services(config: Dict[str, Any], config_name: str)
                     defaulted[service]["selected_network"] = "".join(to_be_referenced)
 
     return defaulted
+
+
+PEER_INFO_MAPPING: Dict[NodeType, str] = {
+    NodeType.FULL_NODE: "full_node_peer",
+    NodeType.FARMER: "farmer_peer",
+}
+
+
+def get_unresolved_peer_infos(service_config: Dict[str, Any], peer_type: NodeType) -> Set[UnresolvedPeerInfo]:
+    peer_info_key = PEER_INFO_MAPPING[peer_type]
+    peer_infos = service_config.get(f"{peer_info_key}s", [])
+    peer_info = service_config.get(peer_info_key)
+    all_peer_infos_raw = [peer_info, *peer_infos] if peer_info is not None else peer_infos
+
+    return set(map(lambda peer: UnresolvedPeerInfo(peer["host"], peer["port"]), all_peer_infos_raw))
```

Comment:
Ok with a comprehension here?

```suggestion
    return {UnresolvedPeerInfo(host=peer["host"], port=peer["port"]) for peer in all_peer_infos_raw}
```

Or, if you take the above adjustment with the effective rename...

```suggestion
    return {UnresolvedPeerInfo(host=peer["host"], port=peer["port"]) for peer in peer_infos}
```